modules/core/src/main/java/org/apache/ignite/internal/processors/cache/transactions/IgniteTxManager.java

```diff
@@ -2240,6 +2244,73 @@ public void sendDeferredAckResponse(UUID nodeId, GridCacheVersion ver) {
     }
 
     /**
+     * Detaches current thread from the transaction.
+     *
+     * @param tx Transaction to be detached.
+     * @return {@code False} if already detached.
+     */
+    public void suspendTx(final GridNearTxLocal tx) throws IgniteCheckedException {
+        assert tx != null;
+        assert !tx.system() && !tx.pessimistic();
+
+        if (!tx.suspendInProgress())
+            throw new IgniteCheckedException("Use suspendTx prohibited. Use tx.suspend() instead.");
+
+        clearThreadMap(tx);
+
+        transactionMap(tx).remove(tx.xidVersion(), tx);
+
+        tx.txTopForSuspension(txTop.get());
+
+        tx.threadId(UNDEFINED_THREAD_ID);
+
+        tx.state(SUSPENDED);
+    }
+
+    /**
+     * Attaches current thread to transaction.
+     *
+     * @param tx Transaction to be attached.
+     */
+    public void resumeTx(GridNearTxLocal tx) throws IgniteCheckedException {
+        assert tx != null;
+        assert !tx.system() && !tx.pessimistic();
+        assert tx.threadId() == UNDEFINED_THREAD_ID;
+        assert !threadMap.containsValue(tx);
+        assert !transactionMap(tx).containsValue(tx);
```

Comment:
As far as I can see other method use transactionMap.
Moreover common method for obtaining map allows some additional checks and actions inside it.
Future logic can be extended so it better to use transactionMap method.